src/pal/pal_ws_lws.c

```diff
@@ -1045,16 +1049,23 @@ static int32_t pal_wsworker_get_proxy_info(
         *port = 0;
         return er_ok;  // No proxy configured, return success
     }
+    port_str = __prx_config_get(prx_config_key_proxy_port, NULL);
     user = __prx_config_get(prx_config_key_proxy_user, NULL);
     pwd = __prx_config_get(prx_config_key_proxy_pwd, NULL);
 
-    buf_len = strlen(proxy) + (user ? strlen(user) : 0 ) + (pwd ? strlen(pwd) : 0)
-        + 2 + 1;  // plus 2 for pwd/user sep and null terminator
+    buf_len = 
+          strlen(proxy) 
+        + (port_str ? strlen(port_str) + 1 : 0)  // Add port sep
+        + (user ? strlen(user) + 1 : 0) // user sep @
+        + (pwd ? strlen(pwd) + 1 : 0) // add pwd sep :
+        + 1;  // plus null term
     
-    *proxy_address = (char*)crt_alloc(buf_len); // Use malloc for lws to free
+    // Use malloc for lws to free
+    *proxy_address = (char*)crt_alloc(buf_len); 
     if (!*proxy_address)
         return er_out_of_memory;
-    memset(*proxy_address, 0, buf_len); // Avoid use of uninitialized memory, issue #27
+
+    *proxy_address = 0;
```

Comment:
Undoing this change fixes the segmentation fault, which I reported a few moments ago.
So the memset for the whole *proxy_address buffer is needed, not just the first character.